src/coreclr/vm/threads.cpp

```diff
@@ -8304,7 +8309,65 @@ BOOL dbgOnly_IsSpecialEEThread()
 
 #endif // _DEBUG
 
+void Thread::StaticInitialize()
+{
+    WRAPPER_NO_CONTRACT;
+
+#ifdef FEATURE_SPECIAL_USER_MODE_APC
+    InitializeSpecialUserModeApc();
+
+    // When CET shadow stacks are enabled, support for special user-mode APCs with the necessary functionality is required
+    _ASSERTE_ALL_BUILDS(__FILE__, !AreCetShadowStacksEnabled() || UseSpecialUserModeApc());
+#endif
+}
+
+#ifdef FEATURE_SPECIAL_USER_MODE_APC
+
+QueueUserAPC2Proc Thread::s_pfnQueueUserAPC2Proc;
+
+static void NTAPI EmptyApcCallback(ULONG_PTR Parameter)
+{
+    LIMITED_METHOD_CONTRACT;
+}
+
+void Thread::InitializeSpecialUserModeApc()
+{
+    WRAPPER_NO_CONTRACT;
+    static_assert_no_msg(OFFSETOF__APC_CALLBACK_DATA__ContextRecord == offsetof(CLONE_APC_CALLBACK_DATA, ContextRecord));
+
+    HMODULE hKernel32 = WszGetModuleHandle(W("kernel32.dll"));
+    if (hKernel32 == nullptr)
+    {
+        hKernel32 = WszLoadLibrary(W("kernel32.dll"));
+        if (hKernel32 == nullptr)
+        {
+            return;
+        }
+    }
+
```

Comment:
```suggestion
    HMODULE hKernel32 = WszLoadLibraryEx(WINDOWS_KERNEL32_DLLNAME_W, NULL, LOAD_LIBRARY_SEARCH_SYSTEM32);
```
I do not think we bother with dual WszGetModuleHandle / WszLoadLibrary lookup anywhere else. I would expect the LoadLibrary path to be fairly optimized for already-loaded libraries.